### PR TITLE
Rename CustomError to Custom

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

Redudent "Error" in `InstructionError::CustomError` and `ProgramError::CustomError`

#### Summary of Changes

Rename it to `Custom`

Fixes #

### CODE REVIEW

**Comment by jackcmay** [comment]
@mvines @jstarry Backward incompatible but would be a nice nit cleanup.  Would also nice to be able to apply this to `InstructionError::CustomError`.  Thoughts?

**Comment by mvines** [comment]
This doesn't worry me, it's source backwards incompatible but no affect to binaries AFAICT

**Comment by jackcmay** [comment]
@mvines, Thoughts on also modifying `InstructionError::CustomError`?

**Comment by mvines** [comment]
wfm

File: sdk/src/instruction.rs [hunk 1]
```diff
@@ -116,7 +116,7 @@ pub enum InstructionError {
     /// by the Solana runtime. A program-specific error may be any type that is represented as
     /// or serialized to a u32 integer.
     #[error("program error: {0:#x}")]
-    CustomError(u32),
+    Custom(u32),
```
Comment (jstarry) [inline sdk/src/instruction.rs:119]: the error message above says "program error", should we rename `Custom` to `Program`?

Comment (jstarry) [reply sdk/src/instruction.rs:119]: Never mind, I think `Custom` makes more sense. Could you update the error message to "custom program error"?

**Pull request merged**